tests/python/test_grammar_compiler.py

```diff
@@ -13,6 +13,7 @@
 from xgrammar.testing import _get_allow_empty_rule_ids
 
 
+@pytest.mark.hf_token_required
```

Comment:
This is a new testing marker I added to flag tests that require a huggingface token with access to the gated models needed in these tests. We skip all these tests at package qualification but I think we still get pretty good coverage. We could run these in CI by passing an HF_TOKEN in an env but then the github runners don't have GPUs so we can't test CUDA paths.